internal/gke/deploy.go

```diff
@@ -169,13 +164,29 @@ func PrepareRollout(ctx context.Context, config CloudConfig, cfg *config.GKEConf
 		buildLocation, config.Project, dockerRepoName)
 
 	fmt.Fprintln(os.Stderr, "Starting the application container build", appImageURL)
+
+	// Start the build in a separate goroutine.
 	buildDone := make(chan struct{})
+	files := []string{localBinary}
+	var goInstall []string
+	if runtime.GOOS == "linux" && runtime.GOARCH == "amd64" {
+		// Use the running weaver-gke tool binary.
+		toolBinPath, err := os.Executable()
+		if err != nil {
+			return nil, nil, err
+		}
+		files = append(files, toolBinPath)
+	} else {
+		// Cross-compile the weaver-gke tool binary inside the container.
+		goInstall = append(goInstall, "github.com/ServiceWeaver/weaver-gke/cmd/weaver-gke@latest")
```

Comment:
Just a clarifying question, will we have to change `@latest` later to pick a version of the deployer compatible with the version of weaver the developer is uing?

Reply:
Yes, I think so. For now, we grab the latest version.